extensions/wikia/BannerNotifications/js/BannerNotifications.js

```diff
@@ -0,0 +1,309 @@
+/* global require, define*/
+/*
+ * Handles the color-coded notification messages that generally appear
+ * at the top of the screen or inside a modal.
+ *
+ * AMD module is defined at the bottom of the 'require' block
+ *
+ * Use like:
+ * new BannerNotification('Content', 'error').show().hide();
+ */
```

Comment:
Might be worth noting from a security perspective that this module doesn't escape content before inserting it into the DOM, so it's only to be used with content that's already considered safe (i.e. no UCG).